crates/openssl/src/tls_stream.rs

```diff
@@ -135,14 +135,17 @@ async fn connect_tls(
             format!("tls error: {}", e),
         )
     })?;
-    let result = tokio_openssl::connect(tls_config, &dns_name, tcp).await;
-
-    result.map_err(|e| {
+    let mut stream = tls_config
+        .into_ssl(&dns_name)
+        .and_then(|ssl| TokioTlsStream::new(ssl, tcp))
+        .map_err(|e| io::Error::new(io::ErrorKind::Other, format!("tls error: {}", e)))?;
```

Comment:
@bluejekyll  I've removed the `unwrap`, but I am not sure what is the best error kind to propagate.

Reply:
This is fine for now. Thanks for making the change. If it occurs, we can make an explicit error type.